libraries/core_libs/network/src/tarcap/packets_handlers/pbft_block_packet_handler.cpp

```diff
@@ -26,64 +29,62 @@ void PbftBlockPacketHandler::process(const PacketData &packet_data, const std::s
   LOG(log_tr_) << "In PbftBlockPacket";
 
   auto pbft_block = std::make_shared<PbftBlock>(packet_data.rlp_[0]);
-  const uint64_t peer_pbft_chain_size = packet_data.rlp_[1].toInt();
```

Comment:
I would not delete this. Are you 100% sure that it pbft_block->getPeriod()-1 is equivalent to this in all possible situations ?

Reply:
Yes, since proposal block period = pbft chain size + 1

Reply:
I dont think you are right. I checked the code and I dont see such strong guarantee. The node1 that sent you the block might get it from node2, which is far ahead of of node1, but node1 just regossip it to you

Reply:
I guarantee this is right. 
When node generate proposal PBFT block, the block period is always pbft chain size + 1
In your example:
1. when node2 send proposal PBFT block B, generate RLP {B is in period 11, chain size 10}, and send to node1
2. when node1 received the RLP {B is in period 11, chain size 10}, node1 doesn't change the RLP and just regossip it to me
3. So in the RLP block period is always PBFT chain size + 1

Reply:
Exactly, node1 received the block from node2 and then just regossips it to you without adjusting the packet. When the packet gets to you, you save the block period from packet as node1 chain size, which is not true. It is node2 chain size !!!

Reply:
You're right. Good catch! Thank you
Fixed